### PR TITLE
fix(npm): detect npm6/7 for npm install

### PR DESCRIPTION
**Pull request by rarkins**
## Changes:

Adds detection for npm 6 vs 7 when binarySource=docker

## Context:

npm@7 just became `latest` and we should install npm@6 for any repo with a v1 lockFile

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added unit tests, or
- [x] Ran on a real repository

### CODE REVIEW

File: lib/manager/npm/post-update/lerna.ts [hunk 1]
```diff
@@ -56,11 +56,11 @@ export async function generateLockFiles(
       cmdOptions = '--ignore-scripts --ignore-engines --ignore-platform';
     } else if (lernaClient === 'npm') {
       let installNpm = 'npm i -g npm';
-      const npmCompatibility = config.constraints?.npm;
+      const npmCompatibility = config.constraints?.npm as string;
       if (validRange(npmCompatibility)) {
-        installNpm += `@${quote(npmCompatibility)}`;
-        preCommands.push(installNpm, 'hash -d npm');
+        installNpm = `npm i -g ${quote(`npm@${npmCompatibility}`)}`;
```
Comment (rarkins) [inline lib/manager/npm/post-update/lerna.ts:61]: This is an unrelated bug fix

**Review by viceice — DISMISSED** [review]
No tests 😕

**Comment by viceice** [comment]
ref #7474

**Review by viceice — DISMISSED** [review]
